src/solvers/dgsem_p4est/subcell_limiters_2d.jl

```diff
@@ -0,0 +1,257 @@
+        i_secondary_start, i_secondary_step = index_to_start_step_2d(secondary_indices[1],
+                                                                     index_range)
+        j_secondary_start, j_secondary_step = index_to_start_step_2d(secondary_indices[2],
+                                                                     index_range)
+
+        i_primary = i_primary_start
+        j_primary = j_primary_start
+        i_secondary = i_secondary_start
+        j_secondary = j_secondary_start
+
+        for node in eachnode(dg)
+            var_primary = u[variable, i_primary, j_primary, primary_element]
+            var_secondary = u[variable, i_secondary, j_secondary, secondary_element]
+
+            var_min[i_primary, j_primary, primary_element] = min(var_min[i_primary,
+                                                                         j_primary,
+                                                                         primary_element],
+                                                                 var_secondary)
+            var_max[i_primary, j_primary, primary_element] = max(var_max[i_primary,
+                                                                         j_primary,
+                                                                         primary_element],
+                                                                 var_secondary)
+
+            var_min[i_secondary, j_secondary, secondary_element] = min(var_min[i_secondary,
+                                                                               j_secondary,
+                                                                               secondary_element],
+                                                                       var_primary)
+            var_max[i_secondary, j_secondary, secondary_element] = max(var_max[i_secondary,
+                                                                               j_secondary,
+                                                                               secondary_element],
+                                                                       var_primary)
+
+            # Increment primary element indices
+            i_primary += i_primary_step
+            j_primary += j_primary_step
+            i_secondary += i_secondary_step
+            j_secondary += j_secondary_step
+        end
+    end
+
+    # Calc bounds at physical boundaries
+    calc_bounds_twosided_boundary!(var_min, var_max, variable, u, t,
+                                   boundary_conditions,
+                                   mesh, equations, dg, cache)
+
+    return nothing
+end
+
+@inline function calc_bounds_twosided_boundary!(var_min, var_max, variable, u, t,
+                                                boundary_conditions::BoundaryConditionPeriodic,
+                                                mesh::P4estMesh{2},
+                                                equations, dg, cache)
+    return nothing
+end
```

Comment:
How does it work for periodic boundaries in the `p4est` solver? Are they already considered (as interfaces)?

Reply:
Yes exactly. Periodic boundaries are considered as interfaces (as for TreeMesh and StructuredMesh).